spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -282,4 +321,85 @@ public SubscriberStub createSubscriberStub() {
 		}
 	}
 
+	/**
+	 * Creates {@link ExecutorProvider}. If a custom executor provider is set then the
+	 * subscriber properties configured through the application.properties file will be
+	 * ignored.
+	 * @param subscriptionName subscription name
+	 * @return executor provider
+	 */
+	public ExecutorProvider getExecutorProvider(String subscriptionName) {
+		if (this.executorProvider != null) {
+			return this.executorProvider;
+		}
+		if (this.pubSubConfiguration != null) {
+			PubSubConfiguration.Subscriber subscriber = this.pubSubConfiguration.getSubscriber(subscriptionName,
+					this.projectId);
+			return getExecutorProviderFromConfigurations(subscriber, subscriptionName);
+		}
+		return null;
+	}
+
+	/**
+	 * Creates {@link ExecutorProvider} for subscriber when Pub/Sub configurations (either
+	 * global/default or subscription-specific) are present.
+	 * @param subscriber subscriber properties
+	 * @param subscriptionName subscription name
+	 * @return executor provider
+	 */
+	ExecutorProvider getExecutorProviderFromConfigurations(PubSubConfiguration.Subscriber subscriber,
+			String subscriptionName) {
+		if (!subscriber.isGlobal()) {
+			return createExecutorProvider(subscriber, subscriptionName);
+		}
+
+		if (this.defaultExecutorProvider != null) {
+			return this.defaultExecutorProvider;
+		}
+		this.defaultExecutorProvider = createExecutorProvider(subscriber, subscriptionName);
+		return this.defaultExecutorProvider;
+	}
+
+	ExecutorProvider createExecutorProvider(PubSubConfiguration.Subscriber subscriber, String subscriptionName) {
+		if (this.executorProviderMap.containsKey(subscriptionName)) {
+			return this.executorProviderMap.get(subscriptionName);
+		}
+		ThreadPoolTaskScheduler scheduler = createThreadPoolTaskScheduler(subscriber, subscriptionName);
+		scheduler.initialize();
+		ExecutorProvider executor = FixedExecutorProvider.create(scheduler.getScheduledExecutor());
```

Comment:
@elefeint I'm wondering why `ThreadPoolTaskScheduler` is being used here, as only `ScheduledExecutor` created under the hood is passed?

Reply:
We tend to prefer Spring abstractions over plain Java ones for convenience and consistency with other Spring modules. `ThreadPoolTaskScheduler` comes with easy customization of thread names and such.

Reply:
Thanks, that makes perfect sense.